packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -951,121 +796,48 @@ class _DialPainter extends CustomPainter {
-    final double innerLabelRadius = radius - labelPadding * 2.5;
-
-    Offset getOffsetForTheta(double theta, _DialRing ring) {
-      double labelRadius;
-      switch (ring) {
-        case _DialRing.outer:
-          labelRadius = outerLabelRadius;
-          break;
-        case _DialRing.inner:
-          labelRadius = innerLabelRadius;
-          break;
-      }
-      return center + Offset(labelRadius * math.cos(theta),
-          -labelRadius * math.sin(theta));
-    }
-
-    final List<CustomPainterSemantics> nodes = <CustomPainterSemantics>[];
+    paintLabels(primaryLabels);
 
-    void paintLabels(List<_TappableLabel> labels, _DialRing ring) {
-      if (labels == null)
-        return;
-      final double labelThetaIncrement = -_kTwoPi / labels.length;
-      final double ordinalOffset = ring == _DialRing.inner ? 12.0 : 0.0;
-      double labelTheta = math.pi / 2.0;
-
-      for (int i = 0; i < labels.length; i++) {
-        final _TappableLabel label = labels[i];
-        final TextPainter labelPainter = label.painter;
-        final double width = labelPainter.width * _semanticNodeSizeScale;
-        final double height = labelPainter.height * _semanticNodeSizeScale;
-        final Offset nodeOffset = getOffsetForTheta(labelTheta, ring) + Offset(-width / 2.0, -height / 2.0);
-        final TextSpan textSpan = labelPainter.text as TextSpan;
-        final CustomPainterSemantics node = CustomPainterSemantics(
-          rect: Rect.fromLTRB(
-            nodeOffset.dx - 24.0 + width / 2,
-            nodeOffset.dy - 24.0 + height / 2,
-            nodeOffset.dx + 24.0 + width / 2,
-            nodeOffset.dy + 24.0 + height / 2,
-          ),
-          properties: SemanticsProperties(
-            sortKey: OrdinalSortKey(i.toDouble() + ordinalOffset),
-            selected: label.value == selectedValue,
-            value: textSpan?.text,
-            textDirection: textDirection,
-            onTap: label.onTap,
-          ),
-          tags: const <SemanticsTag>{
-            // Used by tests to find this node.
-            SemanticsTag('dial-label'),
-          },
-        );
-        nodes.add(node);
-        labelTheta += labelThetaIncrement;
-      }
-    }
+    final Paint selectorPaint = Paint()
+      ..color = accentColor;
+    final Offset focusedPoint = getOffsetForTheta(theta);
+    const double focusedRadius = _labelPadding - 4.0;
+    canvas.drawCircle(centerPoint, 4.0, selectorPaint);
+    canvas.drawCircle(focusedPoint, focusedRadius, selectorPaint);
+    selectorPaint.strokeWidth = 2.0;
+    canvas.drawLine(centerPoint, focusedPoint, selectorPaint);
 
-    paintLabels(primaryOuterLabels, _DialRing.outer);
-    paintLabels(primaryInnerLabels, _DialRing.inner);
+    // Add a dot inside the selector but only when it isn't over the labels.
+    // This checks that the selector's theta is between two labels. A remainder
+    // between 0.1 and 0.45 indicates that the selector is roughly not above any
```

Comment:
Does this work correctly if font changes or if textScaleFactor > 1?

Reply:
it does run into problems at higher text scale with the dot overlapping the text unfortunately.  I think this is fine though, the existing Android time picker has this problem.